### DISCUSSION TITLE
Broken Pipe / Error in I/O

### DISCUSSION THREAD

**Discussion by qdmt** [outdoorbits/little-backup-box#194] [Q&A] [answered]
Not sure if this is voltage related, but getting the following error. Red light is maintained on RPi 4B/4gb (indicating sufficient voltage) and target is on powered USB hub. Error begins when source is plugged in, whether via 2nd USB port or into the powered hub. It will attempt 5 times then shutdown.

10:09:52
| Ready
| Insert source
| Target o.k.
| size: 1000M
| used: 999M, 100%

10:10:01
| IP (offline):
| 10.41.0.1
| Ready
| Insert source
| Target o.k.

10:10:14
DEVICE_CHOSEN_IDENT='--uuid 564B-8C6A' ('/dev/sdb1') prepared to mount at '/media/source', exfat

10:10:14
| mounting:
| USB-source
| IP (offline):
| 10.41.0.1
| Ready

10:10:14
mount_device RESULT=--uuid 564B-8C6A

10:10:15
| Source o.k.
| Working...
| size: 102G
| used: 209M, 1%
| FS: exfat

10:10:15
Files to sync from folder '/media/source': 12

10:10:16
| USB storage 
|  > USB storage 
| 0 of 12
| Time: ?
| 0% ____________________

10:10:16
| USB storage 
|  > USB storage 
| 0 of 12
| Time: ?
| comparing...

10:10:16
/var/www/little-backup-box/tmp/sync.log:
>>>>>

2023/02/11 10:10:15 [6326] building file list
2023/02/11 10:10:15 [6326] .d..t...... ./
2023/02/11 10:10:15 [6326] .d..t...... DCIM/100NCZ_6/
2023/02/11 10:10:15 [6326] rsync: [sender] write error: Broken pipe (32)
<<<<<

10:10:17
Files missing: 12 files not synced.

10:10:17
No abnormal conditions have been detected.

10:10:17
Exception: 11 11=Error in file I/O

10:10:17
No abnormal conditions have been detected.

**Comment by outdoorbits** [comment]
Can you please post the output of Tools > Repair > Check for this partition?

**Reply by qdmt** [reply]
On the target SDA's:

sda4:
17:14:42
> sudo fsck /dev/sda4:
fsck from util-linux 2.36.1

sda3:
17:15:51
> sudo umount /dev/sda3:
umount: /dev/sda3: not mounted.

17:15:51
> sudo fsck /dev/sda3:
fsck from util-linux 2.36.1

sda2:
17:16:15
> sudo umount /dev/sda2:
umount: /dev/sda2: not mounted.

17:16:15
> sudo fsck /dev/sda2:
fsck from util-linux 2.36.1
e2fsck 1.46.2 (28-Feb-2021)
e2fsck: need terminal for interactive repairs

sda1:
17:16:34
> sudo umount /dev/sda1:
umount: /dev/sda1: not mounted.

17:16:34
> sudo fsck /dev/sda1:
fsck from util-linux 2.36.1
fsck.fat 4.2 (2021-01-31)
There is no label in boot sector, but there is volume label 'SYSTEM' stored in root directory
1) Copy volume label from root directory to boot sector
2) Remove volume label from root directory
[12?q]? 

sda:
17:17:08
> sudo fsck /dev/sda:
fsck from util-linux 2.36.1
e2fsck 1.46.2 (28-Feb-2021)
/dev/sda is in use.
e2fsck: Cannot continue, aborting.

**Comment by outdoorbits** [comment, marked as answer]
Your storage is divided into four partitions? Please keep in mind, lbb will use the first partition only. As you see, sda2 seems to have a problem ' e2fsck: need terminal for interactive repairs' and there is something wrong about sda1: 'There is no label in boot sector, but there is volume label 'SYSTEM' stored in root directory'.
I would suggest to fix all partition problems (maybe by format), again check for errors and try to backup again.

**Reply by qdmt** [reply]
Good call - was a quick format, partitions are legacy. Will clean up and try again.

**Reply by qdmt** [reply]
That was it - works a lot smoother. Worked the first time, then powered off to test the sync to see if it would recognize it's already synced - and it has. Thank you!
